### PR TITLE
Add test cases for FhirOperator.kt

### LINKED ISSUE #1075: Implementation of $measure-evaluate API for Population/Practitioner Measure Reporting
**User story**
As an implementer I can use the $measure-evaluate to generate a MeasureReport

Given a set of pre-loaded patients, run the $measure-evaluate operation to generate a MeasureReport
-- On a single patient, the $measure-evaluate will return a MeasureReport with a value of 1 or 0 - **DONE**
-- For a group of patients, the $measure-evaluate will return a MeasureReport - **PENDING/**, TODO  - to add sample test data

**Background**

- [x] Initial #657 works for generation of a measure report and Test case accepts a single subject based on the FHIR spec
- [x] The API should accept the subject parameter, for both a single and group of patients, such that  the measure will be calculated for all subjects that meet the requirements of the measure. If subject is specified, the measure will only be calculated for the referenced subject(s). 
- [x] The API should accept the Practitioner parameter to allow computation of a Measure based on the Practitioner for which the measure will be calculated against. If specified, the measure will be calculated only for subjects that have a primary relationship to the identified practitioner
- [x] Test if we can use other subject apart from Charity-Otala - https://github.com/google/android-fhir/issues/1071 
- [ ] https://github.com/google/android-fhir/issues/1107

More [here](https://www.hl7.org/fhir/operation-measure-evaluate-measure.html)

**Sample code snippet**

```
fun evaluateMeasure(
    url: String,
    start: String,
    end: String,
    reportType: String,
    subject: String
  ): MeasureReport {
    return measureProcessor.evaluateMeasure(
      url,
      start,
      end,
      reportType,
      subject,
      null,
      null,
      null,
      null,
      null,
      null
    )
  }
```

### PR DESCRIPTION
**Pull request by RaaziaTarique**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1075

**Description**
Clear and concise code change description. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Testing

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [ ] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: .github/ci-gradle.properties [hunk 1]
```diff
@@ -17,4 +17,4 @@
  org.gradle.daemon=false
  org.gradle.parallel=true
  org.gradle.jvmargs=-Xmx5120m
- org.gradle.workers.max=2
+ org.gradle.workers.max=4
```
Comment (ekigamba) [inline .github/ci-gradle.properties:20]: Revert this change here

File: workflow/build.gradle.kts [hunk 2]
```diff
@@ -50,6 +50,14 @@ android {
 
   sourceSets { getByName("test").apply { resources.setSrcDirs(listOf("testdata")) } }
 
+  tasks.withType<Test>().configureEach {
```
Comment (ekigamba) [inline workflow/build.gradle.kts:53]: You can also revert the changes here

**Review by ekigamba — CHANGES_REQUESTED** [review]

File: workflow/build.gradle.kts (hunk 2 above)
Comment (RaaziaTarique) [reply workflow/build.gradle.kts:53]: if I revert these changes them, it'll throw Java Heap Exception locally.

**Comment by Tarun-Bhardwaj** [comment]
@RaaziaTarique  , CI issue is fixed. Could you please check if you're still facing it?
CC: @f-odhiambo

File: workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorPopulationTest.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    val measureReportJSON =
+      FhirContext.forR4().newJsonParser().encodeResourceToString(measureReport)
+
+    assertThat(MeasureReport.MeasureReportStatus.COMPLETE).isEqualTo(measureReport.status)
+    assertThat(MeasureReport.MeasureReportType.SUMMARY).isEqualTo(measureReport.type)
+    assertThat("2019-01-01").isEqualTo(DateType(measureReport.period.start).localDate.toString())
+    assertThat("2021-12-31").isEqualTo(DateType(measureReport.period.end).localDate.toString())
+    assertThat(DateType(Date()).localDate).isEqualTo(DateType(measureReport.date).localDate)
+
+    assertThat(measureReportJSON).isNotNull()
```
Comment (ekigamba) [inline workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorPopulationTest.kt:87]: Can we assert the output of the report to make sure that this works as expected

File: workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt [hunk 4]
```diff
@@ -75,31 +77,17 @@ class FhirOperatorTest {
       )
     val measureReportJSON =
       FhirContext.forR4().newJsonParser().encodeResourceToString(measureReport)
+    assertThat(MeasureReport.MeasureReportStatus.COMPLETE).isEqualTo(measureReport.status)
+    assertThat(MeasureReport.MeasureReportType.INDIVIDUAL).isEqualTo(measureReport.type)
+    assertThat(DateType(Date()).localDate).isEqualTo(DateType(measureReport.date).localDate)
+    assertThat("2020-01-01").isEqualTo(DateType(measureReport.period.start).localDate.toString())
+    assertThat("2020-01-31").isEqualTo(DateType(measureReport.period.end).localDate.toString())
+    assertThat("Patient/charity-otala-1").isEqualTo(measureReport.subject.reference)
```
Comment (ekigamba) [inline workflow/src/test/java/com/google/android/fhir/workflow/FhirOperatorTest.kt:85]: Are there other values that we can assert that possibly provide 'documentation' on how this function is supposed to work

**Review by ekigamba — CHANGES_REQUESTED** [review]

**Pull request closed without merging**